cupy/_creation/ranges.py

```diff
@@ -254,6 +254,104 @@ def logspace(start, stop, num=50, endpoint=True, base=10.0, dtype=None,
     return _core.power(base, y).astype(dtype)
 
 
+def _geomspace_scalar(start, stop, num=50, endpoint=True, dtype=None):
+    """
+    Scalar implementation for geomspace using NumPy for calculation.
+    Assumes start and stop are Python scalars.
+
+    Args:
+        start: Start of the interval.
+        stop: End of the interval.
+        num: Number of elements.
+        endpoint (bool): If ``True``, the stop value is included as the last
+            element. Otherwise, the stop value is omitted.
+        dtype: Data type specifier. It is inferred from the start and stop
+            arguments by default.
+    Returns:
+        cupy.ndarray: The 1-D array of ranged values.
+
+    .. seealso:: :func:`numpy.geomspace`
+    """
+    if start == 0 or stop == 0:
+        raise ValueError('Geometric sequence cannot include zero')
+
+    np_result = numpy.geomspace(start, stop, int(
+        num), endpoint=endpoint, dtype=dtype)
+
+    # return object as cupy array
+    cp_result = cupy.asarray(np_result)
+
+    return cp_result
+
+
+def geomspace(start, stop, num=50, endpoint=True, dtype=None, axis=0):
+    """
+    Return numbers spaced evenly on a log scale (a geometric progression).
+
+    This is similar to `logspace`, but with endpoints specified directly.
+    Each output sample is a constant multiple of the previous.
+
+    Args:
+        start: Start of the interval.
+        stop: End of the interval.
+        num: Number of elements.
+        endpoint (bool): If ``True``, the stop value is included as the last
+            element. Otherwise, the stop value is omitted.
+        dtype: Data type specifier. It is inferred from the start and stop
+            arguments by default.
+        axis (int):  The axis in the result to store the samples.  Relevant
+            only if start or stop are array-like.  By default ``0``, the
+            samples will be along a new axis inserted at the beginning.
+            Use ``-1`` to get an axis at the end.
+    Returns:
+        cupy.ndarray: The 1-D array of ranged values.
+
+    .. seealso:: :func:`numpy.geomspace`
+    """
+
+    if num < 0:
+        raise ValueError("Number of samples, `num`, must be non-negative.")
+
+    # Fast track if parameters are scalars
+    if cupy.isscalar(start) and cupy.isscalar(stop):
+        return _geomspace_scalar(start, stop, num, endpoint, dtype)
+
+    start = cupy.asanyarray(start)
+    stop = cupy.asanyarray(stop)
+    if cupy.any(start == 0) or cupy.any(stop == 0):
+        raise ValueError('Geometric sequence cannot include zero')
+
+    dt = cupy.result_type(start, stop, float(num), cupy.zeros((), dtype=dtype))
```

Comment:
`cupy.zeros((), dtype=dtype)` unnecesarilly allocates the memory. Could you avoid this?